consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/ibftmessagedata/IbftUnsignedPrePrepareMessageData.java

```diff
@@ -59,4 +62,22 @@ public Block getBlock() {
   public int getMessageType() {
     return TYPE;
   }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    IbftUnsignedPrePrepareMessageData that = (IbftUnsignedPrePrepareMessageData) o;
+    return Objects.equals(block, that.block)
+        && Objects.equals(roundIdentifier, that.roundIdentifier);
```

Comment:
Should this equals just check the properties this class added and use super.equals for the parts defined in the superclass?

Reply:
Ditched this and made a custom equality function (can be dangerous using 'equals' in a heirarchy of classes.